shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -902,8 +902,11 @@ private native void nativeInvokePlatformMessageEmptyResponseCallback(
   // TODO(mattcarroll): differentiate between channel responses and platform responses.
   @UiThread
   public void invokePlatformMessageResponseCallback(
-      int responseId, @Nullable ByteBuffer message, int position) {
+      int responseId, @NonNull ByteBuffer message, int position) {
     ensureRunningOnMainThread();
+    if (!message.isDirect()) {
```

Comment:
does it also fix flutter/flutter#81875?

Reply:
Half of the problem, there were 2.  The `position` problem and the direct buffer problem.

Reply:
do we want to remove the ability to set `null`?

Reply:
Yep, check out where this function is actually used: https://github.com/gaaclarke/engine/blob/7219fc0932a238cee073bb20f19ccd8f8729d022/shell/platform/android/io/flutter/embedding/engine/dart/DartMessenger.java#L160:L161   There is a separate function you are supposed to use if you want to send null.

Reply:
sg